sdk/identity/identity/src/credentials/azurePipelinesServiceConnectionCredential.ts

```diff
@@ -0,0 +1,165 @@
+import { credentialLogger } from "../util/logging";
+import { checkTenantId } from "../util/tenantIdUtils";
+import {
+  createDefaultHttpClient,
+  createHttpHeaders,
+  createPipelineRequest,
+} from "@azure/core-rest-pipeline";
+import { AzurePipelinesServiceConnectionCredentialOptions } from "./azurePipelinesServiceConnectionCredentialOptions";
+
+const credentialName = "AzurePipelinesServiceConnectionCredential";
+
+const logger = credentialLogger(credentialName);
+
+export class AzurePipelinesServiceConnectionCredential implements TokenCredential {
+  private client: ClientAssertionCredential | undefined;
+  private serviceConnectionId: string | undefined;
+
+  /**
+   * AzurePipelinesServiceConnectionCredential supports Microsoft Entra Workload ID on Kubernetes.
+   *
+   * @param options - The identity client options to use for authentication.
+   */
+  constructor(
+    clientId: string,
+    tenantId: string,
+    serviceConnectionId: string,
+    options?: AzurePipelinesServiceConnectionCredentialOptions
+  ) {
+    checkTenantId(logger, tenantId);
+    logger.info(
+      `Invoking AzurePipelinesServiceConnectionCredential with tenant ID: ${tenantId}, clientId: ${clientId} and service connection id: ${serviceConnectionId}`
+    );
+
+    if (clientId && tenantId && serviceConnectionId) {
+      // Ensure all system env vars are there to form the request uri for OIDC token
+      this.ensurePipelinesSystemVars();
+      const oidcRequestUrl = `${process.env.SYSTEM_TEAMFOUNDATIONCOLLECTIONURI}${process.env.SYSTEM_TEAMPROJECTID}/_apis/distributedtask/hubs/build/plans/${process.env.SYSTEM_PLANID}/jobs/${process.env.SYSTEM_JOBID}/oidctoken?api-version=7.1-preview.1&serviceConnectionId=${this.serviceConnectionId}`;
+      const systemAccessToken = `${process.env.SYSTEM_ACCESSTOKEN}`;
+      logger.info(
+        `Invoking ClientAssertionCredential with tenant ID: ${tenantId}, clientId: ${clientId} and service connection id: ${serviceConnectionId}`
+      );
+      this.client = new ClientAssertionCredential(
+        tenantId,
+        clientId,
+        this.requestOidcToken.bind(this, oidcRequestUrl, systemAccessToken),
+        options
+      );
+    }
+  }
+
+  /**
+   * Authenticates with Microsoft Entra ID and returns an access token if successful.
+   * If authentication fails, a {@link CredentialUnavailableError} will be thrown with the details of the failure.
+   *
+   * @param scopes - The list of scopes for which the token will have access.
+   * @param options - The options used to configure any requests this
+   *                TokenCredential implementation might make.
+   */
+  public async getToken(
+    scopes: string | string[],
+    options?: GetTokenOptions
+  ): Promise<AccessToken | null> {
+    if (!this.client) {
+      const errorMessage = `${credentialName}: is unavailable. tenantId, clientId, and either tokenFilePath or serviceConnectionId are required parameters. 
```

Comment:
This error message needs updating and some nuance - tokenFilePath is not applicable here, and this is not a Workload Identity Federation (or it is but a very specific type of it right?)

Reply:
Yes I updated that. Thanks!